motionplan/rrt.go

```diff
@@ -12,15 +15,15 @@ const (
 	defaultPlanIter = 20000
 )
 
-type rrtParallelPlanner interface {
+type rrtMotionPlanner interface {
 	motionPlanner
-	rrtBackgroundRunner(context.Context, spatialmath.Pose, []referenceframe.Input, *rrtParallelPlannerShared)
+	initRRTSolutions(context.Context, spatialmath.Pose, []referenceframe.Input) *rrtPlanReturn
 }
 
-type rrtParallelPlannerShared struct {
-	maps            *rrtMaps
-	endpointPreview chan node
-	solutionChan    chan *rrtPlanReturn
+type rrtParallelMotionPlanner interface {
```

Comment:
I'm a little wary of doing this abstraction.

We're definitely going to eventually have non-RRT planners, so distinguishing RRT and non-RRT planners makes sense.

But it's entirely possible we'll never have non-parallel RRT planners. In which case we're just adding an extra layer of abstraction for no reason.

Reply:
I agree.  I meant to call this `parallelMotionPlanner` and have it only depend on `motionPlanner` about to push up some code illustrating this